econplayground/main/views.py

```diff
@@ -591,3 +593,168 @@ def get_success_url(self):
             extra_tags='safe')
 
         return reverse('topic_list', kwargs={'cohort_pk': self.cohort.pk})
+
+
+class AssignmentListView(LoginRequiredMixin, ListView):
+    model = Assignment
+    template_name = 'main/assignment_list.html'
+
+    def dispatch(self, request, *args, **kwargs):
+        if not user_is_instructor(request.user):
+            url = '/accounts/login/'
+            return HttpResponseRedirect(url)
```

Comment:
There is no need to do this redirection manually as you are already using LoginRequiredMixin in this view.